H₀ (Null Hypothesis): 
The completion rate of the Test group is less than or equal to the completion rate of the Control group 
(p_test ≤ p_control).

H₁ (Alternative Hypothesis): 
The completion rate of the Test group is greater than the completion rate of the Control group 
(p_test > p_control).

Significance level: α = 0.05

In [ ]:
#setup
import pandas as pd
import numpy as np
from statsmodels.stats.proportion import proportions_ztest
from scipy.stats import ttest_ind

# load data
df = pd.read_csv("/Users/hinahaq/Downloads/data_file_project5/final_test_data.csv")
print("Loaded:", df.shape)
df.head()


Loaded: (317235, 7)


,client_id,visitor_id,visit_id,process_step,date_time,source,group_test
0,3561384,451664975_1722933822,100012776_37918976071_457913,confirm,2017-04-26 13:22:17,pt1,test
1,3561384,451664975_1722933822,100012776_37918976071_457913,confirm,2017-04-26 13:23:09,pt1,test
2,7338123,612065484_94198474375,100019538_17884295066_43909,start,2017-04-09 16:20:56,pt1,test
3,7338123,612065484_94198474375,100019538_17884295066_43909,step_1,2017-04-09 16:21:12,pt1,test
4,7338123,612065484_94198474375,100019538_17884295066_43909,step_2,2017-04-09 16:21:21,pt1,test


In [ ]:
# Order matters: [Test, Control] for "larger" test
counts = [21731, 16046]  # Number of successes 
nobs = [33157, 30910] #Total Trials

stat, p_z = proportions_ztest(counts, nobs, alternative='larger')
print(f"Correct Z-test (Test > Control):")
print(f"Z = {stat:.3f}, p = {p_z:.6f}")
print("Reject H0:", "YES" if p_z < 0.05 else "NO")

Correct Z-test (Test > Control):
Z = 35.040, p = 0.000000
Reject H0: YES


**RESULT**
A one-sided proportions z-test showed that the Test group achieved a significantly higher completion rate than the Control group (65.5% vs. 51.9%; z = 35.04, p < 0.000001), indicating that the new design led to a statistically significant improvement in process completion.
